src/realm/object-store/impl/deep_change_checker.hpp

```diff
@@ -43,7 +44,7 @@ class RealmCoordinator;
 struct CollectionChangeInfo {
     TableKey table_key;
     ObjKey obj_key;
-    ColKey col_key;
+    Path path;
```

Comment:
Storing a path like this only works in the simplest of situations. It's only a stable identifier if it's one element long; if it's actually nested the indexes will become invalid as soon as there's an insertion or removal on the parent list at an index ahead of the observed path.

This design would require reviving the very complicated old logic for updating tracked indices for row removals that we used before we had stable keys, and would bring along with it all of the problems that lead to us introducing stable keys. Alternatively we could try to find some way to introduce stable keys for lists (which would probably have significant overhead?).

Reply:
Damn - I missed that part. We actually have stable keys for collections. We will have to see how those can be used here.

Reply:
this should now be ok.